starter-core/src/main/java/io/micronaut/starter/feature/database/DataJdbc.java

```diff
@@ -64,8 +66,13 @@ public void apply(GeneratorContext generatorContext) {
                 .groupId("io.micronaut.data")
                 .artifactId("micronaut-data-jdbc")
                 .compile());
+
         DatabaseDriverFeature dbFeature = generatorContext.getRequiredFeature(DatabaseDriverFeature.class);
-        generatorContext.getConfiguration().addNested(getDatasourceConfig(dbFeature));
+        Map<String, Object> datasourceConfig = getDatasourceConfig(dbFeature);
```

Comment:
Instead of adding it in `getDataSourceConfig` and then removing it, do you think it would be better to pass the generator context to `getDataSourceConfig` and don't add the config if a Migration feature is present there?

Reply:
I have mixed feelings about this. I avoid changing a API unless there's a compelling reason to do so. If I change as you suggested, then knowledge of the `Migration` feature is pushed up to the `DataFeature` interface, but that's relevant to just a few implementations, two of which already override the method. `DataJpa` overrides the `getDataSourceConfig` in its implementation, to never add that conditional config. `DataR2dbc` overrides it as well since the config is different for it; thus, the `Migration` logic has to be tested in the implementation class anyway.

I have a pushed a commit that makes your suggested change. See what you think. Thoughts @sdelamo?